subiquity/server/runner.py

```diff
@@ -17,11 +17,23 @@
 from contextlib import suppress
 import os
 import subprocess
-from typing import List, Optional
+from typing import Any, List, Optional
 
 from subiquitycore.utils import astart_command
 
 
+class ProcessWithArgs:
```

Comment:
This is hard to look at to convince myself, with code review only, that it is not introducing a logic change.
 Especially as there is some typing handling already in place.  Do you feel like existing tests cover the users of ProcessWithArgs well?

Reply:
Yeah, I kind of agree with you... It felt like a good option when I wrote it but now I wonder if that was really a good thing. Maybe I should drop this patch and think again how to improve this.

The types hints in place currently are sort of broken for this use case ; because we rely on a member variable that we add dynamically, but isn't present in we just instantiate a `asyncio.subprocess.Process`. This is the error I'm trying to address:
```
subiquity/server/runner.py:75: error: "Process" has no attribute "args"
subiquity/server/runner.py:82: error: "Process" has no attribute "args"
subiquity/server/runner.py:85: error: "Process" has no attribute "args"
```

Reply:
If we have a test, I'm OK with it.
If we don't have a test, I'd like to see one or pull this hunk out and bring it back later with a test.

Reply:
So this touches on something I've been a bit concerned about anyway: curtin.py has an API where you call start_curtin_command and get a Command object, which then has methods like .wait() and so on. But runner.py has an API where you get a runner and call start() to get a, well, thing, that you then pass back to other runner methods. I wonder if runner.py having an API more like curtin.py would be better. Viewed one way, this is taking what you've done here and going a bit further with it (and will less attribute forwarding please).

If you decide this is a good idea, it would be good for another branch I think.

Reply:
Okay I'm dropping this patch for now. Thank you both for your input!